tensorrt_llm/_torch/models/checkpoints/base_weight_mapper.py

```
@@ -123,49 +185,102 @@ def rename_by_params_map(self, params_map: dict[str, str],
+        If simple string renaming rules used in `rename_by_params_map` does not satisfy the
+        need, call this function for a more custom rewrite of the checkpoint weight dict.
+
+        Args
+        - weights: Mapping[str, torch.Tensor], full checkpoint/state-dict weight
+          tensors keyed by checkpoint name.
+
+        Returns
+        - weights: Mapping[str, torch.Tensor], preprocessed weight dict to pass to
+          the child-module loader.
         """
         ...
```

Ruff (0.16.1)

[warning] `BaseWeightMapper.preprocess_weights` is an empty method in an abstract base class, but has no abstract decorator

(B027)